esFrontLine/auth.py

```diff
@@ -0,0 +1,116 @@
+from mohawk import Receiver
+import random
+import logging
+import time
+
+
+logger = logging.getLogger('esFrontLine')
+
+class AuthException(Exception):
+    '''
+    Exception occuring during the authentication protocol
+    '''
+
+
+class HawkAuth(object):
+    '''
+    Authenticate users using HAWK protocol
+    '''
+    def __init__(self):
+        self.users = []
+        self.seen = {}
+
+    def load_users(self, users):
```

Comment:
I like this method; it prevents errors caused by bad datastructures.  But rather than confirming the users with the correct structure, it would prefer a message saying why the user was rejected, and what was expected.  Maybe I would go so far as to raise an exception and let the whole program exit with error.